src/analyses/base.ml

```diff
@@ -2059,6 +2059,53 @@ struct
     let st: store = ctx.local in
     let gs = ctx.global in
     let desc = LF.find f in
+    let memory_copying dst src =
+      let dest_a, dest_typ = addr_type_of_exp dst in
+      let src_lval = mkMem ~addr:(Cil.stripCasts src) ~off:NoOffset in
+      let src_typ = eval_lv (Analyses.ask_of_ctx ctx) gs st src_lval
+                    |> AD.get_type in
+      (* when src and destination type coincide, take value from the source, otherwise use top *)
+      let value = if typeSig dest_typ = typeSig src_typ then
```

Comment:
Previously `Memcpy` used the same type lookup pattern for both `src` and `dst`, but now they're different. What's the reason for changing that?

Reply:
That was just a simplification as the operations performed to compute `dest_a` and `dest_typ` originally are just the same as `addr_type_of_exp`. Nothing has changed semantically. I do not use `addr_type_of_exp` for `src` as it would unnecessarily compute an address not needed